backend/backend/common/room.py

```diff
@@ -0,0 +1,41 @@
+import uuid
+
+from cached_property import cached_property
+
+from .player import Player
+
+
+class Room:
+    def __init__(self):
+        self._players = {}
+
+    @property
+    def players(self):
+        return self._players
+
+    @property
+    def leader(self):
+        leader = next(filter(lambda player: player.leader
+                             , self._players.values()))
+        return leader
+
+    @cached_property
+    def room_id(self):
+        return uuid.uuid4().hex[0:6]
+
+    def add_player(self, player: Player):
+        self._players.update({player.id: player})
+
+    def update_leader(self, player_id, new_leader_id):
+        player = self._players.get(player_id)
+        new_leader = self._players.get(new_leader_id)
+        if not player or not new_leader:
+            return "One of the given ids does not exist"
+
+        if not player.leader:
+            return "Sender id not room admin"
```

Comment:
In python it is "Better forgiveness than permission" I think you should raise custom exception in each case, thus open the ability to utilize the handler on exception provided by fastapi

Reply:
doen